porch/pkg/registry/porch/secret.go

```diff
@@ -44,7 +65,144 @@ func (r *secretResolver) ResolveCredential(ctx context.Context, namespace, name
 		return repository.Credential{}, fmt.Errorf("cannot resolve credentials in a secret %s/%s: %w", namespace, name, err)
 	}
 
+	for _, resolver := range r.resolverChain {
```

Comment:
Is there a reason to introduce the indirection here, vs `if secret.Type == basic { doBasic() }` etc

Reply:
I think we can simply the auth machinery here. Shared my mental model in a previous comment.